apis/workloads/v1alpha1/lifecycle_recorder.go

```diff
@@ -24,6 +24,7 @@ func CommonEventFromConsole(ctx string, eventKind events.EventKind, csl *Console
 
 // +kubebuilder:object:generate=false
 type LifecycleEventRecorder interface {
+	GetIdForConsole(string, string, time.Time) string
```

Comment:
I don't feel like this quite belongs inside of this interface. I **do** think we should create a top-level function that does this in the package, wrapping the one we implement in the `pkg/workloads/console/events` package with helpers accepting a console object.

I'm happy to be persuaded but my thoughts are that we should keep the event recorder interface fairly lean and as agnostic to the implementation details as possible, whilst adding these helpers to the implementation itself.
I don't believe the event recorder interface should have much of an idea about what an event ID entails.